browser-test/src/applicant/northstar_upsell.test.ts

```diff
@@ -60,6 +62,54 @@ test.describe('Upsell tests', {tag: ['@northstar']}, () => {
     await validateApplicationSubmittedPage(
       page,
       /* expectRelatedProgram= */ true,
+      /* expectedDownloadApplicationLink= */ true,
+      applicantQuestions,
+    )
+
+    await test.step('Validate screenshot and accessibility', async () => {
+      await validateScreenshot(
+        page,
+        'upsell-north-star-download',
+        /* fullPage= */ true,
+        /* mobileScreenshot= */ true,
+      )
+    })
+
+    await validateAccessibility(page)
+
+    await test.step('Validate that user can return to the homepage without logging in', async () => {
+      await applicantQuestions.clickBackToHomepageButton()
+      // Expect the login dialog did not appear, so the user should already see the homepage
+      await page.waitForURL('**/programs')
+      // Expect the user is still logged in
+      await expect(page.getByRole('banner')).toContainText(
+        'Logged in as testuser@example.com',
+      )
+    })
+  })
+
+  test('view application submitted page while logged in (no application download link)', async ({
```

Comment:
Question(blocking): Instead of duplicating two large test cases, would it be better to have a smaller test case that has the sole purpose of checking that the download button is not visible? That would allow us to delete a lot of duplicated code

Reply:
yes i am ok with that also -- the chances that removing a single link breaks other parts is tiny.